pilot/pkg/config/memory/monitor.go

```diff
@@ -63,20 +66,25 @@ func NewBufferedMonitor(store model.ConfigStore, bufferSize int) Monitor {
 	}
 
 	return &configstoreMonitor{
+		running:  atomic.NewBool(false),
 		store:    store,
 		handlers: handlers,
 		eventCh:  make(chan ConfigEvent, bufferSize),
 	}
 }
 
 func (m *configstoreMonitor) ScheduleProcessEvent(configEvent ConfigEvent) {
-	m.eventCh <- configEvent
+	if m.running.Load() {
```

Comment:
couldn't there still be a race here? Seems like we could still potentially write to a closed channel.

Reply:
Let me add some more tests here so we don't have to speculate

Reply:
Yes, fixed. Thanks!